syncd/scripts/syncd_init_common.sh

```diff
@@ -124,74 +124,188 @@ config_syncd_cisco_8000()
+                         echo "        $line" >> $to_file
+                         merged_cnt+=1
+                      else
+                         echo "Keep the config $(grep $sedline $to_file) in $to_file"
+                      fi
                    fi
-               fi
-             done < $COMMON_CONFIG_BCM
-             echo "# End of chip common properties" >> $file
-         done
-         echo "Merging $PLT_CONFIG_BCM with $COMMON_CONFIG_BCM, merge files stored in $CONFIG_BCM"
-      fi
-
-      #sync the file system
-      sync
-
-      # copy the final config.bcm and sai.profile to the shared folder for 'show tech'
-      cp -f /tmp/sai.profile /var/run/sswsyncd/
-      cp -f /tmp/*.bcm /var/run/sswsyncd/
-
-      if [ -f "/tmp/sai.profile" ]; then
-          CMD_ARGS+=" -p /tmp/sai.profile"
-      elif [ -f "/etc/sai.d/sai.profile" ]; then
-          CMD_ARGS+=" -p /etc/sai.d/sai.profile"
-      else
-          CMD_ARGS+=" -p $HWSKU_DIR/sai.profile"
-      fi
+                else
+                   echo "        $line" >> $to_file
+                   merged_cnt+=1
+                fi
+            fi
+        fi
+    done < $from_file
 
-    else
+    if [ $merged_cnt -gt 0 ]; then
+         sed -i "/# Start of/a \    global:" $to_file
+         sed -i "/# Start of/a \  0:" $to_file
+         sed -i "/# Start of/a \bcm_device:" $to_file
+         sed -i "/# Start of/a \---" $to_file
+    fi
+    echo "# End of $message" >> $to_file
+    if [ $merged_cnt -gt 0 ]; then
+       sed -i "/# End of/i \..." $to_file
+    fi
+    echo "Merged $from_file to $to_file"
+}
+
+config_syncd_bcm()
+{
+    PLATFORM_COMMON_DIR=/usr/share/sonic/device/x86_64-broadcom_common
+    PLT_CONFIG_BCM=""
+    PLT_CONFIG_YML=""
+    PLT_SAI_PROFILE=$(find $HWSKU_DIR -name 'sai.profile')
+    readline=$(grep SAI_INIT_CONFIG_FILE $PLT_SAI_PROFILE)
+    if [ ${readline: -3} == "bcm" ]; then
+       PLT_CONFIG_BCM=${readline#*=}
+    elif [ ${readline: -3} == "yml" ]; then
+       PLT_CONFIG_YML=${readline#*=}
+    fi
+    
+    if [ ! -z "$PLT_CONFIG_BCM" ] && [ -f $PLATFORM_DIR/common_config_support ] ; then
+       cp -f $HWSKU_DIR/*.config.bcm /tmp
+       cp -f /etc/sai.d/sai.profile /tmp
+       CONFIG_BCM=$(find /tmp -name '*.bcm')
+       SAI_PROFILE=$(find /tmp -name 'sai.profile')
+       sed -i 's+/usr/share/sonic/hwsku+/tmp+g' $SAI_PROFILE
+
+       #Get first three characters of chip id
+       readline=$(grep '0x14e4' /proc/linux-kernel-bde)
+       chip_id=${readline#*0x14e4:0x}
+       chip_id=${chip_id::3}
+       COMMON_CONFIG_BCM=$(find $PLATFORM_COMMON_DIR/x86_64-broadcom_${chip_id} -maxdepth 1 -name '*.bcm')
+
+       if [ -f $PLATFORM_COMMON_DIR/x86_64-broadcom_${chip_id}/*.bcm ]; then
+          for file in $CONFIG_BCM; do
+             merge_config_bcm_files $file $COMMON_CONFIG_BCM "chip common properties"
+          done
+          echo "Merging $PLT_CONFIG_BCM with $COMMON_CONFIG_BCM, merge files stored in $CONFIG_BCM"
```

Comment:
For 7260 hwsku, no config.bcm, just config.bcm.j2.
The merged file doesn't exist, and caused syncd can't boot up.

```
2024 Nov 14 05:02:27.495014 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd Merging /etc/sai.d/config.bcm with /usr/share/sonic/device/x86_64-broadcom_common/x86_64-broadcom_b97/broadcom-sonic-th2.config.bcm, merge files stored in 
2024 Nov 14 05:02:27.505237 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd cp: cannot stat '/tmp/*.bcm': No such file or directory
```

Reply:
So far the script does search the sai.profile to find platform config.
The following copied escaped from the fix.

Can you try the following change for line#250 to see if works ? I will fix it in the PR.

cp -f $HWSKU_DIR/*.config.bcm /tmp => cp -f $PLT_CONFIG_BCM /tmp

Geans